EventTicketingSystem.DataAccess/Services/BookingSeatRepository.cs

```diff
@@ -0,0 +1,86 @@
+﻿using EventTicketingSystem.DataAccess.Interfaces;
+using EventTicketingSystem.DataAccess.Models.Context;
+using EventTicketingSystem.DataAccess.Models.Entities;
+using EventTicketingSystem.DataAccess.Models.Enums;
+using Microsoft.EntityFrameworkCore;
+
+namespace EventTicketingSystem.DataAccess.Services
+{
+    internal class BookingSeatRepository : IBookingSeatRepository
+    {
+        private readonly DatabaseContext _context;
+
+        public BookingSeatRepository(DatabaseContext context)
+        {
+            _context = context;
+        }
+        
+        public async Task<ICollection<BookingSeat>> GetSeats(int bookingId)
+        {
+            return await _context.BookingSeats
+                .Include(bs => bs.EventSeat)
+                .ThenInclude(es => es.Event)
+                .Where(bs => bs.BookingId == bookingId)
+                .ToListAsync();
+        }
+
+        public async Task AddSeat(int bookingId, int eventSeatId, int offerId)
+        {
+            var offer = await _context.Offers.FindAsync(offerId);
```

Comment:
looks like  bussines logic as part of data access only

Reply:
Done